signature/policy_config_test.go

```diff
@@ -152,35 +152,41 @@ func TestDefaultPolicyPath(t *testing.T) {
 	const rootPrefix = "/root/prefix"
 	tempHome := t.TempDir()
 	userDefaultPolicyPath := filepath.Join(tempHome, userPolicyFile)
-
+	tempsystemdefaultpath := filepath.Join(tempHome, systemDefaultPolicyPath)
 	for _, c := range []struct {
-		sys             *types.SystemContext
-		userfilePresent bool
-		expected        string
+		sys                  *types.SystemContext
+		userfilePresent      bool
+		expected             string
+		createSystemTempFile bool
+		expectedError        string
 	}{
 		// The common case
-		{nil, false, systemDefaultPolicyPath},
+		{nil, false, tempsystemdefaultpath, true, ""},
 		// There is a context, but it does not override the path.
-		{&types.SystemContext{}, false, systemDefaultPolicyPath},
+		{&types.SystemContext{}, false, tempsystemdefaultpath, true, ""},
 		// Path overridden
-		{&types.SystemContext{SignaturePolicyPath: nondefaultPath}, false, nondefaultPath},
+		{&types.SystemContext{SignaturePolicyPath: nondefaultPath}, false, nondefaultPath, false, ""},
```

Comment:
If this is testing the override, it should be overriding _something_, i.e. the system file should be present.

---

In _all_ of the following, pre-existing, tests, I think the system file should be present, so that we test which of the various _plausible_ options are chosen.

Reply:
Does this mean that for all test cases that is present should have this system file should be present ??

Reply:
Yes, all test cases from this one on (except for the newly added test, of course). On typical systems, a system policy file is always present, so that’s a more representative test.

(Alternatively, we could test both situations, but that is probably overkill.)

Reply:
> Yes, all test cases from this one on (except for the newly added test, of course). On typical systems, a system policy file is always present, so that’s a more representative test.

I assumed that path is not present. Now i got clarity. I have updated `systemfilepresent` to `true` .